### DISCUSSION TITLE
LR1110 APP key?

### DISCUSSION THREAD

**Discussion by Lozeonjyu** [Lora-net/LoRaMac-node#1042] [Q&A]
Hi,

I was trying to test under TTN but only found Dev EUI, Join EUI and element pin under se-identity.h

What about the other keys like app key and how do i achieve them? 

Thanks

**Comment by mluis1** [comment]
When we get some more time we will try to re-work the LR1110 provisioning in order to ease its setup.

In the meantime you can refer to the answers provided on issue #984 for further details.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Reply by Lozeonjyu** [reply]
Thanks @mluis1 will do so in the future.

I checked through the posts does that mean I use the soft-se for provisioning? What would be the use case for lr1110-se?

Thanks

**Reply by mluis1** [reply]
You can use either `soft-se` or `lr1110-se`.

`lr1110-se` is more secure as all cryptographic operations take place in the radio hardware.

The only difference is that for `lr1110-se` the keys must be provisioned by a different way.
The provided patch example was a way to write the provisioning data in a secure way.

From latest comments it looks that the patch is not fully functional. However, it gives an idea on how to do the job.
